core/valor_core/managers.py

```diff
@@ -0,0 +1,352 @@
+) -> pd.DataFrame:
+    """Checks to see if a dataframe is None before attempting a concatenation. Handles pandas warning about not using pd.concat on empty dataframes."""
+
+    if not df1.empty and (df2 is not None):
+        df1 = pd.concat(
+            [df1, df2],
+            ignore_index=True,
+        )
+    elif df1.empty and (df2 is not None):
+        df1 = df2
+
+    return df1
+
+
+@dataclass
+class ValorDetectionManager:
+    """
+    Manages the evaluation of object detection predictions against groundtruths.
+
+    Attributes
+    ----------
+    datum_uids : set
+        A set of unique identifiers for the data samples.
+    label_map : dict
+        A mapping from one label schema to another.
+    convert_annotations_to_type : enums.AnnotationType or None
+        The target annotation type to convert the data to.
+    metrics_to_return : list[enums.MetricType]
+        A list of metrics to calculate during the evaluation.
+    iou_thresholds_to_compute : list[float]
+        A list of IoU thresholds to compute metrics for.
+    iou_thresholds_to_return : list[float]
+        A list of IoU thresholds to return metrics for.
+    recall_score_threshold : float
+        The score threshold for recall calculations.
+    pr_curve_iou_threshold : float
+        The IoU threshold used for precision-recall curve calculation.
+    pr_curve_max_examples : int
+        The maximum number of examples to include in the precision-recall curve.
+    joint_df : pd.DataFrame
+        A DataFrame containing merged groundtruth and prediction data with calculated IoU.
+    detailed_joint_df : pd.DataFrame
+        A DataFrame containing detailed data for precision-recall curves.
+    unique_groundtruth_labels : dict[set[tuple[str, str]], set[str]]
+        A dictionary mapping labels to unique groundtruth annotation IDs.
+    unique_prediction_labels : set[tuple[str, str]]
+        A set of unique labels present in the predictions.
+    unique_annotation_ids : set[int]
+        A set of unique annotation IDs across groundtruth and prediction data.
+    """
+
+    datum_uids: set = field(default_factory=set)
+    label_map: dict[schemas.Label, schemas.Label] = field(default_factory=dict)
+    convert_annotations_to_type: enums.AnnotationType | None = None
+    metrics_to_return: list[enums.MetricType] = field(
+        default_factory=lambda: [
+            enums.MetricType.AP,
+            enums.MetricType.AR,
+            enums.MetricType.mAP,
+            enums.MetricType.APAveragedOverIOUs,
+            enums.MetricType.mAR,
+            enums.MetricType.mAPAveragedOverIOUs,
+        ]
+    )
+    iou_thresholds_to_compute: list[float] = field(
+        default_factory=lambda: [round(0.5 + 0.05 * i, 2) for i in range(10)]
+    )
+    iou_thresholds_to_return: list[float] = field(
+        default_factory=lambda: [0.5, 0.75]
+    )
+    recall_score_threshold: float = field(default_factory=lambda: 0.0)
```

Comment:
nit: field has a arg called default 

ex. `field(default=0.0)`

Reply:
I would prefer to use this over a lambda when possible.

Reply:
thanks for the suggestion! changed this for all non-collections